accepted/future-releases/small-features-21Q1/feature-specification.md

```diff
@@ -0,0 +1,109 @@
+
+If type arguments are allowed and omitted, the types are inferred from the types of the arguments to the constructor, as for any other constant invocation. This already happens (checked in VM with `dart:mirrors`), so no change is necessary.
+
+## Allow generic function types as type arguments
+
+The language disallows generic function types as type arguments.
+
+```dart
+List<T Function<T>(T)> idFunctions; // INVALID
+var callback = [foo<T>(T value) => value]; // Inferred as above, then invalid.
+```
+
+We remove that restriction, so a type argument *can* be a generic function type.
+
+This requires no new syntax, and in some cases only the removal of a single check. There might be some platforms where the implementation currently assumes that generic function types cannot occur as the value of type variables (an proof-of-concept attempt hit an assert in the VM). Such assumptions will need to be flushed out with tests and fixed.
+
+Because we already infer `List<T Function<T>(T)>` in the code above, this change will not affect type inference, it will just make the inferred type not be an error afterwards.
+
+We do not expect the removal of this restriction to affect the feasibility of type inference. After all, it's already possible to have a generic function type occurring covariantly in a type argument, like `List<T Function<T>(T) Function()>`.
+
+## Allow `>>>` as overridable operator
+
+We reintroduce the `>>>` operator where it originally occurred in the Dart grammar (it's`\gtgtgt`):
+
+```latex
+<shiftOperator> ::= `\ltlt'
+  \alt `\gtgtgt'
+  \alt `\gtgt'
+```
+
+Because this makes `>>>` an `<operator>` and a `<binaryOpartor>`, it directly enables `#>>>` to be written a `Symbol` literal, and it allows declaring `operator >>>` as an instance member with a single positional argument.
+
+Further, the `Symbol` constructor must accept the string `">>>"` as argument and create a symbol equal to `#>>>` (identical if `const` invoked).
+
+Some tests have already been committed, and the feature is currently under the `triple-shift` experiment flag.
+
+Very little actual change is expected since the `>>>` operator behaves equivalently to `>>` and `<<`, so the same code paths should apply as soon as we are past lexical analysis.
+
+When the operator has been enabled, we'll quickly (in the same dev-release series) introduce:
+
+```dart
+int operator >>>(int shift);
+```
+
+on the `int` class, which will work similarly to `>>`, but will zero-extend instead of sign-extend. 
+
+At that point, the `>>>` operator on `int` must be **valid in constant and potentially constant expressions**, so `1 << 3` is a compile-time constant expression with the value `8`, and `const C(int x) : y = 1 << x;` is valid (although potentially throwing if `x > 63`) as a constant constructor.
```

Comment:
Should the example be updated to describe `>>>`?

Reply:
It probably should, yes.